src/software/gui/robot_diagnostics/widgets/robot_diagnostics_gui.cpp

```diff
@@ -78,34 +78,82 @@ RobotDiagnosticsGUI::createDirectControlPrimitiveFromUI()
 {
     auto direct_control_primitive_msg = std::make_unique<TbotsProto::Primitive>();
 
+    setWheelControlPrimitive(std::move(direct_control_primitive_msg));
```

Comment:
Generally, when we want to pass the pointed-to object for a `unique_ptr` around, it's done with a reference instead of passing the actual `unique_ptr` around like this. The helper functions can accept a `TbotsProto::Primitive&` instead. Code structure looks good otherwise.